test/OpenTelemetry.Tests/TestSelfDiagnosticsConfigRefresher.cs

```diff
@@ -6,16 +6,16 @@
 
 namespace OpenTelemetry.Tests;
 
-internal class TestSelfDiagnosticsConfigRefresher(Stream stream = null) : SelfDiagnosticsConfigRefresher
+internal class TestSelfDiagnosticsConfigRefresher(Stream? stream = null) : SelfDiagnosticsConfigRefresher
 {
-    private readonly Stream stream = stream;
+    private readonly Stream? stream = stream;
 
     public bool TryGetLogStreamCalled { get; private set; }
 
     public override bool TryGetLogStream(int byteCount, [NotNullWhen(true)] out Stream stream, out int availableByteCount)
     {
         this.TryGetLogStreamCalled = true;
-        stream = this.stream;
+        stream = this.stream!;
         availableByteCount = 0;
         return true;
     }
```

Comment:
This method states `stream` is not null when the method returns true.
But I don't see that check ever being performed. 

- I think instead the out param needs to be `Stream?`
- remove the null-forgiving operator `!` on line 18
- change line 20 to `return this.stream != null`

Reply:
fixed in b0cbbe775bfa80d9a901c168ecdfeebbb1db0dd0